src/send_test_notification.py

```diff
@@ -0,0 +1,171 @@
+import os
+import xmltodict
+import json
+import config
+from send_status import get_target_url
+from dlc.ticket_notification_handler import TicketNotificationHandler
+
+from codebuild_environment import (
+    get_codebuild_project_name,
+    get_codebuild_project_id,
+    get_codepipeline_url,
+    get_cloudwatch_url,
+)
+
+
+def get_pytest_output():
+    """
+    Get pytest output from file.
+    """
+    pytest_result_directory = os.path.join(os.getcwd(), "test")
+    # get all xml files in test directory
+    files = [
+        os.path.join(pytest_result_directory, f)
+        for f in os.listdir(pytest_result_directory)
+        if f.endswith(".xml")
+    ]
+    # parse xml files and save it to list
+    pytest_output_dict = {}
+    if files:
+        for f in files:
+            with open(f, "r") as xml_file:
+                pytest_output_dict[f] = xmltodict.parse(xml_file.read())
+
+    return pytest_output_dict
+
+
+def get_test_details(name):
+    test_type = os.getenv("TEST_TYPE")
+    test_name = name.split("[")[0]
+    if "ec2" in test_type.lower():
+        repo_instance_name = name.split("[")[1].replace("]", "")
+
+        instance_name = repo_instance_name.split("-")[-1]
+        ecr_image = repo_instance_name.replace(f"-{instance_name}", "")
+
+        return test_name, ecr_image, instance_name
+    else:
+        return test_name, None, None
+
+
+def get_dlc_images(build_context):
+    if build_context == "PR":
+        return os.getenv("DLC_IMAGES")
+    if build_context == "MAINLINE":
+        test_env_file = os.path.join(
+            os.getenv("CODEBUILD_SRC_DIR_DLC_IMAGES_JSON"), "test_type_images.json"
+        )
+        with open(test_env_file) as test_env:
+            test_images = json.load(test_env)
+        for dlc_test_type, images in test_images.items():
+            if dlc_test_type == "sanity":
+                return " ".join(images)
+        raise RuntimeError(f"Cannot find any images for in {test_images}")
+
+
+def get_platform_execution_details(build_context):
+    platform_details = {}
+    codebuild_name = get_codebuild_project_name()
+
+    platform_details["platform_info"] = {}
+    platform_details["platform_info"]["build_context"] = build_context
+    platform_details["platform_info"]["dlc_images"] = get_dlc_images(build_context)
+    platform_details["platform_info"]["test_type"] = os.getenv("TEST_TYPE")
+    platform_details["platform_info"]["codebuild_name"] = codebuild_name
+    platform_details["platform_info"]["codebuild_id"] = get_codebuild_project_id()
+    platform_details["platform_info"]["codebuild_url"] = get_target_url(codebuild_name)
+    platform_details["platform_info"]["cloudwatch_logs_url"] = get_cloudwatch_url(codebuild_name)
+    if build_context == "PR":
```

Comment:
Might want to add these conditionals at the top, with an else statement raising an error that there is no build context.

Reply:
done